src/RageSoundMixBuffer.cpp

```diff
@@ -24,18 +31,22 @@ void RageSoundMixBuffer::SetWriteOffset( int iOffset )
 	m_iOffset = iOffset;
 }
 
-void RageSoundMixBuffer::Extend( unsigned iSamples )
+void RageSoundMixBuffer::Extend(unsigned iSamples)
 {
-	const unsigned realsize = iSamples+m_iOffset;
+	const std::uint64_t realsize = static_cast<std::uint64_t>(iSamples) + m_iOffset;
```

Comment:
I agree it's a little weird, but `m_iOffset` is implicitly converted to `uint64_t` in this statement, so it works out. 

I wanted `realsize` to be `uint64_t` since the original code has no protections against overflow if the sum of `iSamples` and `m_iOffset` is larger than what an `unsigned` can store.